src/downloadmanager.cpp

```diff
@@ -2076,10 +2097,14 @@ void DownloadManager::nxmRequestFailed(QString gameName, int modID, int fileID,
 void DownloadManager::downloadFinished(int index)
 {
   DownloadInfo *info;
-  if (index)
+  if (index > 0)
```

Comment:
Any reason for this change (and the one below)?

Reply:
`downloadFinished` is used in two ways:
- as a function, accepting an integer argument,
- a a signal callback.

Currently, `downloadFinished` has a bug, when it does not run for the first download in the list (index 0), because of the logical mistake here 0==false https://github.com/ModOrganizer2/modorganizer/blob/qt6/src/downloadmanager.cpp#L2079

When it runs with downloadFinished(0), `if (index)` skips execution and jumps to detecting the signal sender, which yeilds nothing, because it is not a signal call. As a result, the first file never sees the finishing routine, causing cascade of errors and discrepancy in interface and actual status. In case of the proposed command, this looks like a duplicating paused file, when user tries to redownload the first file.

The first idea was to add `if (index >= 0)`, but then the signal callback `info = findDownload(this->sender(), &index)` never fires, because during the callback index is also 0.

So I implemented detection of the first row as a fallback after all paths are exhausted.

I expected opposition on this fix and added the additional check in `startDownload`, catching possible downloadFinished(-1) call, which is an invalid value, causing the `no download index` warning. If the fix is accepted, the `startDownload` probably becomes redundant.